### ISSUE TITLE
Feature Request - Ability to use a plantuml server instead of a local .jar

### ISSUE THREAD

**Issue by Mneric** [doxygen/doxygen#9105]
Hello,

I think it would be usefull to be able to generate the plantuml graphs using an external server (deployed using https://github.com/plantuml/plantuml-server) instead of using a local planuml.jar.

In a company, this would be usefull to ensure that everybody uses the same plantuml generation.  
This would also ease the usage of doxygen as the download and copy of the .jar would not be needed anymore.

Thank you!

**Comment by albert-github** [comment]
Some remarks and questions:

> In a company, this would be useful to ensure that everybody uses the same plantuml generation.

- this can lead to problems when the company prohibits to use external connections, though with with MathJax we also can use an external source though this is during the display of the pages (the user can also refer to a locally installed version of MathJax). The normal usage of the plantuml.jar should still remain possible as well.
- "using the same plantuml version", how did you ensure that everybody uses the same doxygen version? I think in the same way the usage of plantuml could be ensured.
- this would require that the data is send to the server, the image is retrieved again from the server and the results can still differ when different users generate the images at different moments (changed version of plantuml on the server). This process could be quite slow, depending on the number of images to be generated and the speed of the internet connection.

**Comment by Mneric** [comment]
> Some remarks and questions:
> 
> > In a company, this would be useful to ensure that everybody uses the same plantuml generation.
> 
> * this can lead to problems when the company prohibits to use external connections, though with with MathJax we also can use an external source though this is during the display of the pages (the user can also refer to a locally installed version of MathJax). The normal usage of the plantuml.jar should still remain possible as well.  

Yes, the normal usage should remain possible. Also, the address of the server should be configurable to target any server.

> * "using the same plantuml version", how did you ensure that everybody uses the same doxygen version? I think in the same way the usage of plantuml could be ensured.  

Yes, it is true. This may not be the best reason.   
Best reason is probably the ease of use if a server is available.

> * this would require that the data is send to the server, the image is retrieved again from the server and the results can still differ when different users generate the images at different moments (changed version of plantuml on the server). This process could be quite slow, depending on the number of images to be generated and the speed of the internet connection.  

In my use case, the server is hosted on the intra network. So the speed should not be an issue.

**Comment by DWIAltonaAnalytics** [comment]
as someone who would like to use a server instead of a local installation here are our reasons:
- we do not want to install java on the machine that generates our documentation
- the server in question is company hosted as no one really wants to send their architecture to someone outside of the company

**Comment by MuppyCwa** [comment]
I second this proposal. Using the [plantuml-server](https://hub.docker.com/r/plantuml/plantuml-server) in a docker environment also makes maintenance easier, upgrade once and deploy, effects are then immediate.

Thanks!

**Comment by albert-github** [comment]
Disclaimer: I have no experience with docker

The question I have:
- how would I send a file with plantuml code to the server and get **all** the files back (a plantuml file can contain multiple images: [example.tar.gz](https://github.com/user-attachments/files/17075517/example.tar.gz))?
- how can I generate files in other formats (e.g. svg)?
